client/components/platform-checkout/save-user/phone-number-input.js

```diff
@@ -123,7 +126,7 @@ const PhoneNumberInput = ( { handlePhoneNumberChange } ) => {
 				) }
 				label={ __( 'Mobile phone number', 'woocommerce-payments' ) }
 				name="platform_checkout_user_phone_field[no-country-code]"
-				value={ inputValue }
+				value={ inputValue.replace( countryCode, '' ) }
```

Comment:
Just noting that this `replace` gets called every time the input field changes, and means that something like `+1` will disappear if the user ever types it in the middle of the phone number field — granted, that would make it an invalid phone number, but it might still be a strange user experience for something the user types to just disappear. Should we instead have the `replace` call happen just once when the input field loads for the first time?